src/jit/morph.cpp

```diff
@@ -11540,7 +11540,20 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             break;
 
         case GT_DIV:
-
+            // Replace "val / dcon" with "val * (1.0 / dcon)" if dcon is a power of two.
+            // Powers of two within range are always exactly represented,
+            // so multiplication by the reciprocal is safe in this scenario
+            if (op2->IsCnsFltOrDbl())
+            {
+                double divisor = op2->AsDblCon()->gtDconVal;
+                if ((typ == TYP_DOUBLE && FloatingPointUtils::hasPreciseReciprocal(divisor)) ||
```

Comment:
In general JIT codebase uses extra paranthesis for clarity: `(typ == TYP_DOUBLE) && FloatingPointUtils::hasPreciseReciprocal(divisor)`

There are similar cases below in the utilities functions.